service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/CaseInitiationControllerTest.java

```diff
@@ -239,42 +205,92 @@ void shouldAttemptGrantAccessToAllLocalAuthorityUsersWhenGrantAccessFailsForSome
     }
 
     @Test
-    void shouldSetPageShowToYesWhenToggleIsEnabled() {
-        given(featureToggleService.isMigrateToManageOrgWarningPageEnabled(anyString())).willReturn(true);
+    void shouldNotValidateWhenToggleIsDisabled() {
+        given(featureToggleService.isAllowCaseCreationForUsersNotOnboardedToMOEnabled(anyString())).willReturn(false);
 
         CaseDetails caseDetails = CaseDetails.builder()
             .data(Map.of("caseName", "title"))
             .build();
 
-        AboutToStartOrSubmitCallbackResponse callbackResponse = postAboutToStartEvent(caseDetails);
+        AboutToStartOrSubmitCallbackResponse response = postMidEvent(caseDetails);
 
-        assertThat(callbackResponse.getData().get(PAGE_SHOW)).isEqualTo("YES");
+        assertThat(response.getErrors()).isEmpty();
     }
 
     @Test
-    void shouldNotSetPageShowWhenToggleIsDisabled() {
-        given(featureToggleService.isMigrateToManageOrgWarningPageEnabled(anyString())).willReturn(false);
-
+    void shouldNotPopulateErrorsWhenToggleIsEnabledAndValidationIsSuccessful() {
         CaseDetails caseDetails = CaseDetails.builder()
-            .data(Map.of("caseName", "title"))
+            .data(Map.of("caseName", "title",
+                "caseLocalAuthority", "example"))
             .build();
 
-        AboutToStartOrSubmitCallbackResponse callbackResponse = postAboutToStartEvent(caseDetails);
+        given(featureToggleService.isAllowCaseCreationForUsersNotOnboardedToMOEnabled(anyString())).willReturn(true);
 
-        assertThat(callbackResponse.getData().get(PAGE_SHOW)).isNotEqualTo("YES");
+        given(localAuthorityValidationService.validateIfUserIsOnboarded())
+            .willReturn(emptyList());
+
+        AboutToStartOrSubmitCallbackResponse actualResponse = postMidEvent(caseDetails);
+
+        assertThat(actualResponse.getErrors().isEmpty());
     }
 
     @Test
-    void shouldRemovePageShowOnAboutToSubmit() {
-        given(featureToggleService.isMigrateToManageOrgWarningPageEnabled(anyString())).willReturn(false);
+    void shouldPopulateErrorsWhenToggleIsEnabledAndValidationFails() {
+        List<String> expectedErrors = new ArrayList<>();
+        expectedErrors.add("Register for an account");
+        expectedErrors.add("You cannot start an online application until you’re fully registered.");
+        expectedErrors.add("Press the back button on your browser to access the link.");
 
         CaseDetails caseDetails = CaseDetails.builder()
-            .data(Map.of("caseName", "title", PAGE_SHOW, "YES"))
+            .data(Map.of("caseName", "title",
+                "caseLocalAuthority", "example"))
+            .build();
+
+        given(featureToggleService.isAllowCaseCreationForUsersNotOnboardedToMOEnabled(anyString())).willReturn(true);
```

Comment:
this is not right, if you mock localAuthorityValidationService, then mocking feature service is pointless, as toggle is inside service. In integration tests we should mock lower level service, e.g OprganisationService and not localAuthorityValidationService

Reply:
habe changed this :)